libraries/chain/include/eosio/chain/webassembly/interface.hpp

```diff
@@ -0,0 +1,581 @@
+         REGISTER_INJECTED_HOST_FUNCTION(_eosio_f32_ne);
+         REGISTER_INJECTED_HOST_FUNCTION(_eosio_f32_lt);
+         REGISTER_INJECTED_HOST_FUNCTION(_eosio_f32_le);
+         REGISTER_INJECTED_HOST_FUNCTION(_eosio_f32_gt);
+         REGISTER_INJECTED_HOST_FUNCTION(_eosio_f32_ge);
+         REGISTER_INJECTED_HOST_FUNCTION(_eosio_f64_add);
+         REGISTER_INJECTED_HOST_FUNCTION(_eosio_f64_sub);
+         REGISTER_INJECTED_HOST_FUNCTION(_eosio_f64_div);
+         REGISTER_INJECTED_HOST_FUNCTION(_eosio_f64_mul);
+         REGISTER_INJECTED_HOST_FUNCTION(_eosio_f64_min);
+         REGISTER_INJECTED_HOST_FUNCTION(_eosio_f64_max);
+         REGISTER_INJECTED_HOST_FUNCTION(_eosio_f64_copysign);
+         REGISTER_INJECTED_HOST_FUNCTION(_eosio_f64_abs);
+         REGISTER_INJECTED_HOST_FUNCTION(_eosio_f64_neg);
+         REGISTER_INJECTED_HOST_FUNCTION(_eosio_f64_sqrt);
+         REGISTER_INJECTED_HOST_FUNCTION(_eosio_f64_ceil);
+         REGISTER_INJECTED_HOST_FUNCTION(_eosio_f64_floor);
+         REGISTER_INJECTED_HOST_FUNCTION(_eosio_f64_trunc);
+         REGISTER_INJECTED_HOST_FUNCTION(_eosio_f64_nearest);
+         REGISTER_INJECTED_HOST_FUNCTION(_eosio_f64_eq);
+         REGISTER_INJECTED_HOST_FUNCTION(_eosio_f64_ne);
+         REGISTER_INJECTED_HOST_FUNCTION(_eosio_f64_lt);
+         REGISTER_INJECTED_HOST_FUNCTION(_eosio_f64_le);
+         REGISTER_INJECTED_HOST_FUNCTION(_eosio_f64_gt);
+         REGISTER_INJECTED_HOST_FUNCTION(_eosio_f64_ge);
+         REGISTER_INJECTED_HOST_FUNCTION(_eosio_f32_promote);
+         REGISTER_INJECTED_HOST_FUNCTION(_eosio_f64_demote);
+         REGISTER_INJECTED_HOST_FUNCTION(_eosio_f32_trunc_i32s);
+         REGISTER_INJECTED_HOST_FUNCTION(_eosio_f64_trunc_i32s);
+         REGISTER_INJECTED_HOST_FUNCTION(_eosio_f32_trunc_i32u);
+         REGISTER_INJECTED_HOST_FUNCTION(_eosio_f64_trunc_i32u);
+         REGISTER_INJECTED_HOST_FUNCTION(_eosio_f32_trunc_i64s);
+         REGISTER_INJECTED_HOST_FUNCTION(_eosio_f64_trunc_i64s);
+         REGISTER_INJECTED_HOST_FUNCTION(_eosio_f32_trunc_i64u);
+         REGISTER_INJECTED_HOST_FUNCTION(_eosio_f64_trunc_i64u);
+         REGISTER_INJECTED_HOST_FUNCTION(_eosio_i32_to_f32);
+         REGISTER_INJECTED_HOST_FUNCTION(_eosio_i64_to_f32);
+         REGISTER_INJECTED_HOST_FUNCTION(_eosio_ui32_to_f32);
+         REGISTER_INJECTED_HOST_FUNCTION(_eosio_ui64_to_f32);
+         REGISTER_INJECTED_HOST_FUNCTION(_eosio_i32_to_f64);
+         REGISTER_INJECTED_HOST_FUNCTION(_eosio_i64_to_f64);
+         REGISTER_INJECTED_HOST_FUNCTION(_eosio_ui32_to_f64);
+         REGISTER_INJECTED_HOST_FUNCTION(_eosio_ui64_to_f64);
+
+         // producer api
+         int32_t get_active_producers(legacy_array_ptr<account_name>) const;
+         REGISTER_LEGACY_HOST_FUNCTION(get_active_producers);
+
+         // crypto api
+         void assert_recover_key(legacy_ptr<const fc::sha256>, legacy_array_ptr<char>, legacy_array_ptr<char>) const;
```

Comment:
First argument type changed from `const fc::sha256&` to pointer to `const fc::sha256`.